File: 198-打家劫舍.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
            
        num1, num2 = nums[-1], 0
        for i in range(len(nums) - 2, -1, -1):
            num1, num2 = max(num1, num2 + nums[i]), num1

        return num1

File: 198-打家劫舍.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if(nums.empty())
            return 0;
        
        int num1 = nums[nums.size() - 1], num2 = 0, num;
        for(int i = nums.size() - 2; i > - 1; i--){
            num = num1;
            num1 = max(num1, num2 + nums[i]);
            num2 = num;
        }

        return num1;
    }
};
